### ISSUE TITLE
Private `to[Member]` member not properly setting state

### ISSUE THREAD

**Issue by TeknoFiend** [microstates/union#9]
The example in the README wasn't working for me so I modified it a bit to extract some info about what is happening.  I _believe_ I'm doing this correctly. 

The initial state value is `"foo"`.  After the `resolve` call the state value appears to be a `String` type representation.

```
const PromiseType = Union({
  Pending: PromiseType => class extends PromiseType {
    resolve(result) {
      return this.toResolved(result);
    }
  },
  Resolved: PromiseType => class extends PromiseType {},
});

let p = PromiseType.Pending.create("foo");

console.log( p.isPending, p.isResolved );  // =>  true false
console.log( p.state );   // =>  foo

p = p.resolve("great!");

console.log( p.isPending, p.isResolved ); // => false true
console.log( p.state );   // =>  
  //  String {length: 3}
  //    0: (...)
  //    1: (...)
  //    2: (...)
  //    3: (...)
  //    4: (...)
  //    length: 3
  //    get 0: ƒ ()
  //    get 1: ƒ ()
  //    get 2: ƒ ()
  //    get 3: ƒ ()
  //    get 4: ƒ ()
  //    __proto__: String
```

I hope this is helpful.

**Comment by cowboyd** [comment]
I think that the problem is that this really ought to be `Any` here instead of `Object` https://github.com/microstates/union/blob/master/src/union.js#L15

Because it's trying to coerce the string into an object which is why you end up with a `String()` instead of a `string`.

Looking back, I think we were trying to be a little too cute with the [ability to assign to the state](https://github.com/microstates/union/blob/master/src/union.js#L66-L75)

This let you do things like say:

```js
this.toOtherState({merged: properties})
```

and it won't overwrite, but will instead merge with any of the properties that are stored in the value. In retrospect, I think this was too clever by half.

**Comment by TeknoFiend** [comment]
Yeah the merging would be awesome.  I want to use an object, actually, and was hoping it would do that (seems natural).  I just tried with an object and ran into a couple more issues, I can open separate tickets if you like.  I tried to make a PR but couldn't get to "working"...

I think this line should be `let value = valueOf(this.value)` : https://github.com/microstates/union/blob/master/src/union.js#L67

And I think the first two conditions (in the next 4 lines) are backwards (I think we want `assign` if it's an object, right?).

Even after making those changes, tho, I wasn't getting a merge or even my new value after the state transition so there must be something else I'm missing.

**Comment by cowboyd** [comment]
I think you're right. Here's the thing though, you can always override the type of the value, in your actual union.

```js
class Base {
  value = Merger;
}

export default Union({Pending: Base => class Pending extends Base {}}, Base);
```

So maybe it makes sense to just make it `Any` and keep it to a simple `set`?

**Comment by TeknoFiend** [comment]
Yes I see what you mean 👍

**Comment by cowboyd** [comment]
Good news is, if it's doesn't work, well we'll trying something else :)
